static/src/javascripts/projects/common/modules/identity/cookierefresh.js

```diff
@@ -1,28 +1,30 @@
 import {
-    isUserLoggedIn,
-    getUserFromApiWithRefreshedCookie,
+	isUserLoggedIn,
+	getUserFromApiWithRefreshedCookie,
+	refreshOktaSession,
 } from 'common/modules/identity/api';
-import { storage } from '@guardian/libs'
+import { storage } from '@guardian/libs';
 
-const shouldRefreshCookie = (
-    lastRefresh,
-    currentTime
-) => {
-    const days30k = 1000 * 86400 * 30; // (as seconds)
-    return !lastRefresh || currentTime > parseInt(lastRefresh, 10) + days30k;
+const shouldRefreshCookie = (lastRefresh, currentTime) => {
+	const days30InSeconds = 1000 * 86400 * 30; // (as seconds)
+	return (
+		!lastRefresh ||
+		currentTime > parseInt(lastRefresh, 10) + days30InSeconds
+	);
 };
 
 const init = () => {
-    const lastRefreshKey = 'identity.lastRefresh';
+	const lastRefreshKey = 'identity.lastRefresh';
 
-    if (storage.local.isAvailable() && isUserLoggedIn()) {
-        const currentTime = new Date().getTime();
-        const lastRefresh = storage.local.get(lastRefreshKey);
-        if (shouldRefreshCookie(lastRefresh, currentTime)) {
-            getUserFromApiWithRefreshedCookie();
-            storage.local.set(lastRefreshKey, currentTime);
-        }
-    }
+	if (storage.local.isAvailable() && isUserLoggedIn()) {
+		const currentTime = new Date().getTime();
+		const lastRefresh = storage.local.get(lastRefreshKey);
+		if (shouldRefreshCookie(lastRefresh, currentTime)) {
```

Comment:
This code is not broken but not sure I understand the logic - we are auto-authenticating users on a rolling basis? Is that secure?

Reply:
This is to avoid people being signed out and then having to sign back in again while at the same time keeping the refresh tokens rotated to reduce the risk of attack.  So it is secure.